streams/src/main/java/org/apache/kafka/streams/processor/internals/Tasks.java

```diff
@@ -136,10 +139,14 @@ private void createActiveTasks(final Map<TaskId, Set<TopicPartition>> activeTask
 
         if (!activeTasksToCreate.isEmpty()) {
             for (final Task activeTask : activeTaskCreator.createTasks(mainConsumer, activeTasksToCreate)) {
-                activeTasksPerId.put(activeTask.id(), activeTask);
-                pendingActiveTasks.remove(activeTask.id());
-                for (final TopicPartition topicPartition : activeTask.inputPartitions()) {
-                    activeTasksPerPartition.put(topicPartition, activeTask);
+                if (stateUpdater != null) {
```

Comment:
This is a meta thought: I think we should consider extracting the creation of tasks out of `Tasks` and into the `TaskManager`, and hence also not include `StateUpdater` into `Tasks`. Maybe we can do that later in a follow-up refactoring if you agree.

Reply:
I agree!

Reply:
Sounds good. I will prepare a PR for this :)